lib/SILOptimizer/Utils/InstOptUtils.cpp

```diff
@@ -1388,6 +1392,91 @@ static bool keepArgsOfPartialApplyAlive(PartialApplyInst *pai,
   return true;
 }
 
+// Given a newly created value @owned val, insert destroy_value at its frontiers
+// when not already found
+void swift::fixupLifetimeOfOwnedValue(SILValue val) {
+  assert(val.getOwnershipKind() == OwnershipKind::Owned);
+  assert(val->getFunction()->hasOwnership());
+
+  SmallVector<SILInstruction *, 4> userSet;
+  for (auto *use : val->getUses()) {
+    if (auto *borrow = dyn_cast<BeginBorrowInst>(use->getUser())) {
+      auto endBorrows = borrow->getEndBorrows();
+      userSet.append(endBorrows.begin(), endBorrows.end());
+    } else {
+      userSet.push_back(use->getUser());
+    }
+  }
+  PointerUnion<SILInstruction *, SILArgument *> def;
+  if (auto *svi = dyn_cast<SingleValueInstruction>(val)) {
+    def = svi;
+  } else if (auto *mvi = dyn_cast<MultipleValueInstruction>(val)) {
+    def = mvi;
+  } else {
+    def = cast<SILPhiArgument>(val);
+  }
+
+  ValueLifetimeAnalysis vla(def, userSet);
+  ValueLifetimeAnalysis::Frontier valFrontier;
+  vla.computeFrontier(valFrontier, ValueLifetimeAnalysis::DontModifyCFG);
+
+#if NDEBUG
```

Comment:
I think you want `#if !NDEBUG` or `#ifndef`. Also please put a commend on the `#endif`.